src/settings/lib/RuleEditor/Rules-hint.js

```diff
@@ -150,18 +182,61 @@ export default function rulesHint(Codemirror, props) {
     }
 
     if (result.length) {
+      sections.forEach(section => {
+        section.header = section.header ? formatMessage({ id: section.header }) : section.header;
+
+        if (section.selectedHintIndex === undefined) {
+          section.selectedHintIndex = -1;
+        }
+      });
+      sections[0].list = result;
+
       return {
-        listDescription: {
-          subheader,
-          list: result,
-        },
+        sections,
         from: Codemirror.Pos(cur.line, start),
         to: Codemirror.Pos(cur.line, end),
-        selectedHint: -1,
         header,
       };
     }
 
     return null;
   });
 }
+
+export function initSubMenuDataFetching(Codemirror, props) {
+  Codemirror.registerHelper('hint', 'getSubMenuData', (cm, options) => {
+    const token = cm.getTokenAt(cm.getCursor());
+    const { state } = token;
+    const { nextApplicable } = state;
+    const { intl: { formatMessage } } = props;
+
+    const {
+      typeMapping,
+      completionLists,
+    } = nextApplicable;
+
+    const result = [];
+
+    if (!typeMapping[options.childSection]) return null;
```

Comment:
Move this under the `type` declaration below and use it like this

```suggestion
    if (!type) return null;
```